source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOResourceResolvers.mm

```diff
@@ -10,25 +10,43 @@
 @implementation ACOResourceResolvers
 {
     NSMutableDictionary<NSString *, NSObject<ACOIResourceResolver> *> *_resolvers;
+    NSMutableDictionary<NSString *, NSNumber *> *_resolversIFMap;
 }
 
 - (instancetype)init
 {
     self = [super init];
     if (self) {
         _resolvers = [[NSMutableDictionary alloc] init];
+        _resolversIFMap = [[NSMutableDictionary alloc] init];
     }
     return self;
 }
 
 - (void)setResourceResolver:(NSObject<ACOIResourceResolver> *)resolver scheme:(NSString *)scheme
 {
     self->_resolvers[scheme] = resolver;
+
+    _resolversIFMap[scheme] = [NSNumber numberWithInt:ACODefaultIF];
+
+    if([resolver respondsToSelector:@selector(resolveImageResource:)]) {
+        _resolversIFMap[scheme] = [NSNumber numberWithInt:ACOImageIF];
+    }
+    // only one IF per scheme is supported and ACRImageViewIF will be chosen
+    // when both respod
```

Comment:
>d [](start = 22, length = 1)

nitpick - typo :)

Reply:
thanks!

---
In reply to: [236838055](https://github.com/Microsoft/AdaptiveCards/pull/2169#discussion_r236838055) [](ancestors = 236838055)